compiler/lib/config.ml

```diff
@@ -195,3 +195,13 @@ let set_target (t : [ `JavaScript | `Wasm ]) =
   | `JavaScript -> Targetint.set_num_bits 32
   | `Wasm -> Targetint.set_num_bits 31);
   target_ := (t :> [ `JavaScript | `Wasm | `None ])
+
+type effects_backend =
+  | Cps
+  | Double_translation
+
+let effects_ : effects_backend option ref = ref None
+
+let effects () = !effects_
```

Comment:
What about doing something similar to `target` above.
```ocaml
type effects_backend =
  | `Cps
  | `Double_translation
  | `Jspi
  | `Not_supported
  | `None

let effects_ : effects_backend option ref = ref `None

let effects () = match !effects_ with
  | `None -> failwith "effect backend was not set"
  | (`Cps | `Double_translation | `Jspi | `Not_supported) as t -> t
```

Reply:
I don't like that `None` currently mean unsupported for jsoo and jspi for wasmoo